src/core/stream/representation/get_needed_segments.ts

```diff
@@ -142,86 +177,132 @@ export default function getNeededSegments({
       }
       return true;
     });
+  let isMemorySaturated = false;
 
-  const segmentsToDownload = availableSegmentsForRange.filter(segment => {
-    const contentObject = objectAssign({ segment }, content);
-
-    // First, check that the segment is not already being pushed
-    if (segmentsBeingPushed.length > 0) {
-      const isAlreadyBeingPushed = segmentsBeingPushed
-        .some((pendingSegment) => areSameContent(contentObject, pendingSegment));
-      if (isAlreadyBeingPushed) {
-        return false;
-      }
-    }
-
-    const { duration, time, end } = segment;
+  return availableSegmentsForRange.filter(segment => {
     if (segment.isInit) {
       return true; // never skip initialization segments
     }
+    if (isMemorySaturated &&
+        bufferLength > MIN_BUFFER_LENGTH &&
+        content.adaptation.type === "video") {
```

Comment:
Even if it works, I find it weird to still do all the same availableBufferSize comparison even when it does not apply (because we're not currently considering a video Adaptation).